controllers/nodefeaturediscovery_state.go

```diff
@@ -50,7 +70,17 @@ func (n *NFD) init(
 	}
 }
 
+// step steps through the list of functions stored in 'n.controls',
+// then attempts to determine if the given resource is Ready or
+// NotReady.
 func (n *NFD) step() error {
+
+	// For each function in n.controls, attempt to check the status of
+	// the relevant resource. If the resource is defined as being
+	// "NotReady," then return an error saying it's not ready. Otherwise,
+	// return the status as being ready, then increment the index for
+	// n.controls so that we can parse the next resource the next time
+	// the step() function is called.
```

Comment:
To me, this is pretty self-evident from the code and I'd be inclined to drop the long-ish comment alltogether.

A one-liner comment before `n.idx = n.idx + 1` could be useful, though, e.g. *// Increment the index to handle the next set of control functions* or smth.

WDYT?